src/directives/ctr-input.ts

```diff
@@ -24,7 +24,17 @@ export class CtrInput {
     private _searchStr = "";
     private _displayStr = "";
 
-		private _handleInputEvents(event: any) {
+    private _keyupActive: boolean = false;
+
+    private handleInputEvents(event: any) {
+
+        // this method is called by both `input` and `keyup` event.
+        // to prevent double-queries, only do the search only once.
+        // `_keyupActive` is set to `true` for `keyup` event and set to `false` for `input` event.
+        if (this._keyupActive) {
```

Comment:
You actually canceled keyup here because you always set it to `true` in `keyup` and to `false` in `input`